### DISCUSSION TITLE
How to create testing environments?

### DISCUSSION THREAD

**Discussion by ChappIO** [directus/directus#7517] [Archive]
I am working on a project where I would like to have separate dev, testing, and production environments. Currently I am doing this by provisioning the database using knex migrations (creating tables, configuring displays, etc...) but it is a quite tedious process.

Is there any better way to do it?

Wat I strive to do is:

1. Develop all changes on my development machine
2. Roll out to testing using CI/CD which will migrate the testing db to the new schema and settings
3. Repeat step 2 but for production.

Update: I did try importing things like the CollectionService from “directus/dist/services” inside the knex migration and that seems to work. Any known reasons why I shouldn’t go this path?

**Comment by benhaynes** [comment]
This happens to be our most requested feature... and it's next on our list! https://github.com/directus/directus/discussions/3891
